test/workload_discovery/unix/kafka.go

```diff
@@ -41,26 +42,36 @@ func RunKafkaTest() error {
 }
 
 func testKafkaVersion(version string) error {
+	// TODO: remove once Java 11 no longer persists
+	if strings.Contains(version, "4.") {
+		cmd := exec.Command("java", "-version")
+		output, _ := cmd.CombinedOutput()
+		if strings.Contains(string(output), "11.") {
+			log.Printf("Skipping Kafka %s - requires Java 17+, found Java 11", version)
+			return nil
+		}
+	}
+
 	defer func() {
 		time.Sleep(util.ShortSleep)
-		exec.Command("sh", "-c", fmt.Sprintf("rm -rf /tmp/%s*", version)).Run()
+		os.RemoveAll("/tmp/kafka")
 	}()
 	// Setup Kafka
 	env := environment.GetEnvironmentMetaData()
 	cmd := exec.Command("./unix/util/scripts", "setup_kafka", version, env.Bucket)
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("failed to setup Kafka %s: %v", version, err)
 	}
-	kafkaDir := fmt.Sprintf("/tmp/%s", version)
+	kafkaDir := fmt.Sprintf("/tmp/kafka/%s", version)
 	time.Sleep(util.MediumSleep)
 
 	// Test NEEDS_SETUP phase
 	cmd = exec.Command("./unix/util/scripts", "spin_up_kafka", kafkaDir, version)
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("failed to start Kafka without JMX: %v", err)
 	}
-	time.Sleep(util.MediumSleep)
-	if err := util.CheckJavaStatusWithRetry("NEEDS_SETUP/JMX_PORT", "Kafka Broker", "KAFKA/BROKER", KafkaPort); err != nil {
+	time.Sleep(util.LongSleep)
```

Comment:
The change from `MediumSleep` to `LongSleep` makes me think this is arbitrary and could be flaky. Could we poll to check if the process is running with a timeout instead?